--- a/luci-app-ssr-plus/Makefile
+++ b/luci-app-ssr-plus/Makefile
@@ -32,12 +32,10 @@ LUCI_DEPENDS:= \
 LUCI_TITLE:=SS/SSR/V2Ray/Trojan/NaiveProxy/Tuic/ShadowTLS/Hysteria/Socks5/Tun LuCI interface
 LUCI_PKGARCH:=all
 LUCI_DEPENDS:= \
+	@(PACKAGE_libustream-mbedtls||PACKAGE_libustream-openssl||PACKAGE_libustream-wolfssl) \
 	+coreutils +coreutils-base64 +dns2socks +dns2tcp +dnsmasq-full +@PACKAGE_dnsmasq_full_ipset +ipset +kmod-ipt-nat \
-	+ip-full +iptables +iptables-mod-tproxy +lua +lua-neturl +libuci-lua +microsocks \
+	+ip-full +iptables +iptables-mod-tproxy +lua +lua-neturl +libuci-lua +luci-lib-ipkg +microsocks \
 	+tcping +resolveip +shadowsocksr-libev-ssr-check +uclient-fetch \
-	+PACKAGE_$(PKG_NAME)_INCLUDE_libustream-mbedtls:libustream-mbedtls \
-	+PACKAGE_$(PKG_NAME)_INCLUDE_libustream-openssl:libustream-openssl \
-	+PACKAGE_$(PKG_NAME)_INCLUDE_libustream-wolfssl:libustream-wolfssl \
 	+PACKAGE_$(PKG_NAME)_INCLUDE_V2ray:curl \
 	+PACKAGE_$(PKG_NAME)_INCLUDE_V2ray:v2ray-core \
 	+PACKAGE_$(PKG_NAME)_INCLUDE_Xray:curl \
@@ -65,23 +63,6 @@ LUCI_DEPENDS:= \
 
 
 define Package/$(PKG_NAME)/config
-select PACKAGE_luci-lib-ipkg if PACKAGE_$(PKG_NAME)
-
 choice
-	prompt "Uclient SSL Lib Selection"
-	default PACKAGE_$(PKG_NAME)_INCLUDE_libustream-openssl
-	
-	config PACKAGE_$(PKG_NAME)_INCLUDE_libustream-mbedtls
-	bool "libustream-mbedtls"
-
-	config PACKAGE_$(PKG_NAME)_INCLUDE_libustream-openssl
-	bool "libustream-openssl"
-
-	config PACKAGE_$(PKG_NAME)_INCLUDE_PACKAGE_libustream-wolfssl
-	bool "libustream-wolfssl"
-endchoice
-
-choice
 	prompt "Shadowsocks Client Selection"
 	default PACKAGE_$(PKG_NAME)_INCLUDE_Shadowsocks_Rust_Client if aarch64 || x86_64
 	default PACKAGE_$(PKG_NAME)_INCLUDE_Shadowsocks_Libev_Client
