--- a/luci-app-fchomo/Makefile
+++ b/luci-app-fchomo/Makefile
@@ -9,7 +9,6 @@ LUCI_DEPENDS:= \
 LUCI_DEPENDS:= \
 	+mihomo \
 	+ca-bundle \
-	+firewall4 \
 	+kmod-inet-diag \
 	+kmod-nft-tproxy \
 	+yq +wget-ssl +coreutils-timeout \
--- a/nikki/Makefile
+++ b/nikki/Makefile
@@ -31,7 +31,7 @@ LUCI_DEPENDS:= \
   CATEGORY:=Network
   TITLE:=A rule based proxy in Go.
   URL:=https://wiki.metacubex.one
-  DEPENDS:=$(GO_ARCH_DEPENDS) +ca-bundle +curl +yq firewall4 +ip-full +kmod-inet-diag +kmod-nft-socket +kmod-nft-tproxy +kmod-tun
+  DEPENDS:=$(GO_ARCH_DEPENDS) +ca-bundle +curl +yq +ip-full +kmod-inet-diag +kmod-nft-tproxy +kmod-tun
   PROVIDES:=nikki mihomo
 endef
 