diff --git a/modules/luci-base/po/zh_Hans/base.po b/modules/luci-base/po/zh_Hans/base.po
index 87436e05c6..dd54ced8fa 100644
--- a/modules/luci-base/po/zh_Hans/base.po
+++ b/modules/luci-base/po/zh_Hans/base.po
@@ -8754,6 +8757,24 @@
 msgid "« Back"
 msgstr "« 后退"
 
+msgid "Source Code"
+msgstr "开源代码"
+
+msgid "Custom Redirect Domain"
+msgstr "自定义挟持域名"
+
+msgid "Define a custom domain name and the corresponding PTR record"
+msgstr "自定义域名对应的IP地址（需要客户端DNS指向本路由）"
+
+msgid "Domain Name"
+msgstr "域名（不带 HTTP(S)://）"
+
+msgid "<abbr title=\"Internet Protocol Version 4\">IPv4</abbr>-Address"
+msgstr "<abbr title=\"互联网协议第 4 版\">IPv4</abbr> 地址"
+
+msgid "Comments"
+msgstr "备注"
+
 msgid "Enables The 802.11k standard provides information to discover the best available access point"
 msgstr "启用 802.11k 自动寻找发现最佳可用接入点的信息"
 
diff --git a/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js b/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
index 45f7b4acae..08fbde85c5 100644
--- a/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
+++ b/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
@@ -78,6 +78,21 @@ return baseclass.extend({
 			);
 		}
 
+		var projectlink = document.createElement('a');
+		projectlink.append('Project OpenWrt');
+		projectlink.href = 'https://github.com/nanchuci';
+		projectlink.target = '_blank';
+
+		var corelink = document.createElement('a');
+		corelink.append('Powered By nanchuci');
+		corelink.href = 'https://github.com/benboguan/Auto-immortalwrt-mt798x';
+		corelink.target = '_blank';
+
+		var sourcelink = document.createElement('placeholder');
+		sourcelink.append(projectlink);
+		sourcelink.append(' / ');
+		sourcelink.append(corelink);
+
 		var fields = [
 			_('Hostname'),         boardinfo.hostname,
 			_('Model'),            boardinfo.model + cpubench.cpubench,
@@ -93,6 +108,7 @@ return baseclass.extend({
 				systeminfo.load[2] / 65535.0
 			) : null,
 			_('CPU usage (%)'),    cpuusage.cpuusage
+			_('Source Code'),      sourcelink
 		];
 
 		if (tempinfo.tempinfo) {
